tests/keras/utils/data_utils_test.py

```diff
@@ -53,5 +64,151 @@ def test_data_utils():
     os.remove('test.txt')
     os.remove('test.zip')
 
+
+"""Enqueuers Tests"""
+
+
+class threadsafe_iter:
+    """Takes an iterator/generator and makes it thread-safe by
+    serializing call to the `next` method of given iterator/generator.
+    """
+
+    def __init__(self, it):
+        self.it = it
+        self.lock = threading.Lock()
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        return self.next()
+
+    def next(self):
+        with self.lock:
+            return next(self.it)
+
+
+def threadsafe_generator(f):
+    """A decorator that takes a generator function and makes it thread-safe.
```

Comment:
This is in a test file, it's just my go-to decorator to handle generators when pickle_safe=False. 
In the tests, I want to test that it works for multiple workers.